src/transformers/pipelines/depth_estimation.py

```diff
@@ -91,6 +91,8 @@ def preprocess(self, image, timeout=None):
         image = load_image(image, timeout)
         self.image_size = image.size
         model_inputs = self.image_processor(images=image, return_tensors=self.framework)
+        if self.framework == 'pt':
+            model_inputs = {k: v.type(self.torch_dtype) if v.dtype == torch.float32 else v for k, v in model_inputs.items()}
```

Comment:
You should be able to do: 
```suggestion
            model_inputs = model_inputs.to(torch_dtype)
```

is there a reason for filtering on inputs which are already in float32?

Reply:
The reason being some image processor can output bboxes see: https://github.com/huggingface/transformers/blob/dcdda5324bcc7a750b5e40e11dd795442204ff27/src/transformers/pipelines/object_detection.py#L109-L111 or things that are supposed to be in integer, and those need not be converted to float else it take extra vram for no reason. Please correct me if I am wrong on this though as I have not benchmarked it.

While for inputs that are in fp32, it means they are supposed to be in floats and thus it is safe to convert to fp16.

Reply:
This is very specific to certain tasks. I wouldn't expect anything like bounding boxes for depth estimation. 

Calling `.to(dtype)` should only convert the tensors which were already floating point types

Reply:
> Calling .to(dtype) should only convert the tensors which were already floating point types

thats something I didnt know. Will change it then.